### PR TITLE
Tgui Jumbo PR: Fixes, CSS refactoring, new components

### LINKED ISSUE #47577: crafting boots back to the main crafting menu when you craft
the crafting menu goes back to the main menu when you craft an item so it is very hard to craft ammunition and anything else that you would need in bulk.

possible solutions: don't go back when crafting, allow mass crafting (x5, x10, etc)

### LINKED ISSUE #47649: New ORM Interface Won't Dispense Alloys
The new TGUI-next interface for the ORM is pretty, but doesn't allow any vending of alloys (materials made by smelting multiple ores together, such as plasteel or plastitanium) whatsoever.

### PR DESCRIPTION
**Pull request by stylemistake**
## About The Pull Request

Fixes #47577
Fixes #47649

Massive refactoring of CSS:

- Massively improve performance of CSS compiler (build time went down 18s -> 7s).
- Reduced size of JS (332KB -> 309KB)
- Reduced size of CSS (65KB -> 44KB)
- Nuke all old colors.
- Use a spectrum-based set of colors.
- Tweak colors for best visual appearance in UIs, use a more robust color math, fonts should be always bright and legible.
- Rounded corners in all UI controls.
- Tweak baselines of UI elements.
- Nuke useless vars, rename all vars to more appropriate names.
- Button icons are now fixed width, so buttons no longer change size when toggling them.

Small refactoring of UIs:

- Created a reusable BeakerContents component.
- Improved table layout of Smart Fridge and Crew Console.
- Code style fixes.

New features:

- `ColorBox` component for rendering a small 1-character wide color box.
- `Dimmer` component for dimming whole window pane for showing important content.
- `AnimatedNumber` render prop is now called with two parameters: `(formattedValue, rawValue) => any`.
- `Table.Row` and `Table.Cell` have a `header` property for declaring table headers.
- `ProgressBar` supports the whole range of colors.

Fixes:

- Crafting UI will no longer reset its state when pressing "Craft".
- Computer Fabricator actually works.

Miscellaneous:

- Updated all dependencies to latest versions. This should fix compatibility with Node v13.
- Reintroduced a custom Babel plugin for multiline string compile-time evaluation, which reduces bundle size by about 5-6KB.
- Cherry picked corejs features to reduce bundle size by further 20KB.
- Node v13 compatibility by removing Fibers package without noticeable slowdown.

<br><details>
<summary><b>LOTS AND LOTS OF SCREENSHOTS</b></summary><br>

image

image

image

image

image

image

image

image

</details>

## Why It's Good For The Game

It's good, mmkay?

## Changelog
:cl:
fix: Personal Crafting UI will no longer reset its state when crafing an item.
fix: Improved table layout of Smart Fridge and Crew Console.
fix: Fix ORM not smelting alloys.
fix: Fix Computer Fabricator.
refactor: Tgui CSS got a massive rework.
/:cl:

### CODE REVIEW

**Comment by TheVekter** [comment]
Did you work out drop down menus or is that still WIP?

**Comment by stylemistake** [comment]
@TheVekter actioninja is currently working on them.

**Comment by HomocidlePancakes** [comment]
Rounded corner are fucking virgin AF. I want my 90° angles back

**Comment by Ohagi-Chan** [comment]
Thank goodness. I got the impression crafting wasn't getting fixed.
https://github.com/tgstation/tgstation/issues/47577

**Comment by stylemistake** [comment]
It was easy enough to fix.

**Review by actioninja — CHANGES_REQUESTED** [review]

File: tgui-next/packages/tgui/interfaces/PersonalCrafting.js [hunk 1]
```diff
@@ -88,98 +88,90 @@ export const PersonalCrafting = props => {
...
-            spin={1}
-          />
-        </Box>
-      </Fragment>
-    );
-  }
+  const craftingRecipes = map((subcategory, category) => {
+    const hasSubcats = ('has_subcats' in subcategory);
+    const firstSubcatName = Object.keys(subcategory)
+      .find(name => name !== 'has_cubcats');
```
[the remark this answers is not in the document]
Comment (stylemistake) [reply tgui-next/packages/tgui/interfaces/PersonalCrafting.js]: it was your typo, and you said it was fine.

**Comment by stylemistake** [comment]
Chad Lummox has added support for NaN and Infinity. We can finally observe fusion glitches in full glory! http://www.byond.com/forum/post/2520870

File: tgui-next/packages/tgui/interfaces/PersonalCrafting.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (actioninja) [reply tgui-next/packages/tgui/interfaces/PersonalCrafting.js]: I don't even remember saying that and you probably shouldn't have listened to me. It should be `has_subcats` in every place it appears.

**Comment by stylemistake** [comment]
This is more or less final. Please test/review, feel free to merge.

**Review by actioninja — DISMISSED** [review]

**Review by SpaceManiac — CHANGES_REQUESTED** [review]
This PR breaks Wine support

First problem:
image

Second problem:
image

**Comment by optimumtact** [comment]
rounded corners eww

**Comment by stylemistake** [comment]
To reduce bikeshedding and start solving real issues, I decided to revert some controversial changes.

### Button rounding is gone.

crab rave

Now click that juicy **MERGE** button!

**Pull request merged**